pydantic/generics.py

```diff
@@ -211,7 +212,7 @@ def iter_contained_typevars(v: Any) -> Iterator[TypeVarType]:
         yield v
     elif hasattr(v, '__parameters__') and not get_origin(v) and lenient_issubclass(v, GenericModel):
         yield from v.__parameters__
-    elif isinstance(v, Iterable):
+    elif isinstance(v, Iterable) and not lenient_issubclass(v, Enum):
```

Comment:
@daviskirk WDYT? Do you think we should improve the check even more to handle other cases?

Reply:
I was almost sure I would miss something here and this is a good example.

The trick here is that the Type itself has to be an iterable for this to fail.  So an Enum class is a good example because it's a weird hybrid of a normal class and an object/singleton kind of thing.  

There might be other classes like this but I don't know them (and to be honest I didn't even know that the enum class was in iterable, so that's a cool new trick I learned).

So I'm fine with this the way it is unless we find a bunch of other examples at which point we might want to think about how to rework this check.

Reply:
Perfect! I had the exact same thought 👍

Reply:
Humm, given the risk of infinite recursion, I'd rather whitelist than blacklist here.

The following seems to pass tests, can either of you think of a case where this might break/change behaviour? If so can we easily add that case to the `isinstance` whitelist too?

```py
DictValues: Type[Any] = {}.values().__class__


def iter_contained_typevars(v: Any) -> Iterator[TypeVarType]:
    """Recursively iterate through all subtypes and type args of `v` and yield any typevars that are found."""
    if isinstance(v, TypeVar):
        yield v
    elif hasattr(v, '__parameters__') and not get_origin(v) and lenient_issubclass(v, GenericModel):
        yield from v.__parameters__
    elif isinstance(v, (DictValues, list)):
        ...
```

Reply:
I think the only reason for DictValues is the `typevars.values()` in line 110, so I'd be fine with changing that to `list(typevars.values())` there and just have the list check here.

Reply:
Lunch break is over. Feel free to modify this PR. Otherwise I'll do the change tonight